api/tests/integration/edge_api/identities/test_edge_identity_featurestates_viewset.py

```diff
@@ -1033,3 +1044,135 @@ def _create_segment_override(
+        feature_state_value=target_feature_state_2_value,
+    )
+
+    target_feature_state_3 = FeatureStateModel(
+        feature=feature_model_3,
+        environment_id=dynamo_enabled_environment,
+        enabled=False,
+    )
+
+    # Add feature states for features 1 and 2 to source identity
+    source_identity.add_feature_override(feature_state=source_feature_state_1)
+    source_identity.add_feature_override(feature_state=source_feature_state_2)
+
+    # Add feature states for features 2 and 3 to target identity.
+    # This feature state must be deleted as part of the cloning process.
+    target_identity.add_feature_override(feature_state=target_feature_state_2)
+    target_identity.add_feature_override(feature_state=target_feature_state_3)
+
+    # Save identities to identities document
+    target_identity_document = target_identity.to_document()
+    source_identity_document = source_identity.to_document()
+
+    flagsmith_identities_table.put_item(Item=target_identity_document)
+    flagsmith_identities_table.put_item(Item=source_identity_document)
+
+    # WHEN
+
+    # Clone feature states from source identity to target identity
+    clone_identity_feature_states_response = admin_client.post(
+        path=featurestates_urls["clone-from-given-identity"].substitute(
+            {"identity_uuid": target_identity.identity_uuid}
+        ),
+        data=json.dumps(
+            obj={"source_identity_uuid": str(object=source_identity.identity_uuid)}
+        ),
+        content_type="application/json",
+    )
+
+    # Get current feature states for fixture target identity
+    get_feature_states_response = admin_client.get(
+        path=featurestates_urls["all"].substitute(
+            {"identity_uuid": target_identity.identity_uuid}
+        ),
+    )
+
+    # THEN
+
+    # Assert API calls responses status are HTTP OK (200)
+    assert clone_identity_feature_states_response.status_code == status.HTTP_200_OK
+    assert get_feature_states_response.status_code == status.HTTP_200_OK
+
+    # Assert target identity contains only the 2 cloned features states
+    response = get_feature_states_response.json()
+
+    # Assert target identity contains only the 2 cloned overridden features states and 1 environment feature state
+    assert len(response) == 3
+
+    # Assert cloned data is correct
+    assert response[0]["feature"]["id"] == feature_1.id
+    assert response[0]["enabled"] == source_feature_state_1.enabled
+    assert response[0]["feature_state_value"] == source_feature_state_1_value
+    assert response[0]["overridden_by"] == "IDENTITY"
+
+    assert response[1]["feature"]["id"] == feature_2.id
+    assert response[1]["enabled"] == source_feature_state_2.enabled
+    assert response[1]["feature_state_value"] == source_feature_state_2_value
+    assert response[1]["overridden_by"] == "IDENTITY"
+
+    assert response[2]["feature"]["id"] == feature_3.id
+    assert response[2]["overridden_by"] is None
```

Comment:
I'd add asserts for `enabled` and `feature_state_value` even if they're not part of this response even testing their non-inclusion would be useful.

Reply:
Definitely not part of the scope of the functionality being tested as far as it is not an identity override is OK, right?

Reply:
Added anyway since in this case is very cheap in all senses.